### DISCUSSION TITLE
Is there a way to get a parent parent's state in a Yup schema?

### DISCUSSION THREAD

**Discussion by npearson72** [react-hook-form/react-hook-form#9999] [Q&A] [answered]
I tried getting an answer from the yup folks, but haven't yet. I also wonder if maybe the `useForm` options might help, so that's why I'm asking here.

Say I have two Yup schemas that I want to concat:

```js
const foo = object({
  foo: object({
    name: string().required()
  })
})

const bar = object({
  bar: object({
    nickname: string().required()
  })
})
```

I then concat these together to produce a final schema:

```js
const schema = foo.concat(bar)

useForm({
  resolver: yupResolver(schema)
})
```

How can I configure the `nickname` field to be invalid if the `name` field is the same value?

I know the `useForm` hook also accepts a `context`. Can that help in any way? I just don't see how I can access a field's parent's parent. Has anyone else encountered this?

**Comment by npearson72** [comment, marked as answer]
Found the solution to this. Looks like it didn't need the extra context that `useForm` configs offer, and was in fact just a Yup thing:

```js
const foo = object({
  foo: object({
    name: string().required()
  })
})

const bar = object({
  bar: object({
    nickname: string().required()
  })
}).test({
  test: (value, ctx) => {
    if (ctx.parent.foo.name !== value.nickname) {
      return true;
    }

    return ctx.createError({
      message: 'Boom!',
      path: 'bar.nickname'
    });
  }
})
```

**Comment by Kazragone** [comment]
thanks, you saved me a thon

> Found the solution to this. Looks like it didn't need the extra context that `useForm` configs offer, and was in fact just a Yup thing:
> 
> ```js
> const foo = object({
>   foo: object({
>     name: string().required()
>   })
> })
> 
> const bar = object({
>   bar: object({
>     nickname: string().required()
>   })
> }).test({
>   test: (value, ctx) => {
>     if (ctx.parent.foo.name !== value.nickname) {
>       return true;
>     }
> 
>     return ctx.createError({
>       message: 'Boom!',
>       path: 'bar.nickname'
>     });
>   }
> })
> ```
